### PR TITLE
[RFC] [DX] [Behat] Traits usage in new behats

### PR DESCRIPTION
**Pull request by lchrusciel**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | yes
| Deprecations?   | no
| Related tickets | 
| License         | MIT
| Doc PR          | 

In order to reduce code duplication we agreed to give traits a try, to check if they would fit our needs

### CODE REVIEW

File: src/Sylius/Behat/Behaviour/ChoosesName.php [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ */
+
+namespace Sylius\Behat\Behaviour;
+
+/**
+ * @author Łukasz Chruściel <lukasz.chreusciel@lakion.com>
+ */
+trait ChoosesName
+{
+    use DocumentAccessor
```
Comment (pamil) [inline src/Sylius/Behat/Behaviour/ChoosesName.php]: Missing `;`

File: src/Sylius/Behat/Behaviour/ChoosesName.php [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Behat\Behaviour;
+
+/**
+ * @author Łukasz Chruściel <lukasz.chreusciel@lakion.com>
```
Comment (Zales0123) [inline src/Sylius/Behat/Behaviour/ChoosesName.php]: Typo in author block :D

File: src/Sylius/Behat/Behaviour/DocumentAccessor.php [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Behat\Behaviour;
+
+use Behat\Mink\Element\DocumentElement;
+
+/**
+ * @author Łukasz Chruściel <lukasz.chreusciel@lakion.com>
```
Comment (Zales0123) [inline src/Sylius/Behat/Behaviour/DocumentAccessor.php]: And here.

**Comment by pamil** [comment]
drinking-anchorman-relaxed-ULyYV5amK2eYM

File: src/Sylius/Behat/Behaviour/ChoosesName.php [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Behat\Behaviour;
```
Comment (pamil) [inline src/Sylius/Behat/Behaviour/ChoosesName.php]: IMHO it should be renamed to Sylius\Behat\Behaviour**s**, as there are many, different behaviours that have nothing in common

Comment (Zales0123) [inline src/Sylius/Behat/Behaviour/ChoosesName.php]: as well as ``Sylius\Component\Core\Model`` should be renamed to ``Sylius\Component\Core\Models``, as there are many model**s** there, that have many different functionalities? ;)

Comment (michalmarcinkowski) [inline src/Sylius/Behat/Behaviour/ChoosesName.php]: We always use singular form in namespaces `Sylius\Behat\Context`, `Sylius\Behat\Page` etc. so IMO we should be consistent and leave `Behaviour`

Comment (pamil) [inline src/Sylius/Behat/Behaviour/ChoosesName.php]: @Zales0123 all right, I've gone too far with these disses :crying_cat_face:

Comment (Zales0123) [inline src/Sylius/Behat/Behaviour/ChoosesName.php]: @pamil don't worry, dark side's traps are tricky ;)

**Comment by michalmarcinkowski** [comment]
:+1:

**Comment by pjedrzejewski** [comment]
:dancer:

**Pull request merged**